qiskit/circuit/instruction.py

```diff
@@ -250,7 +250,7 @@ def inverse(self):
             QiskitError: if the instruction is not composite
                 and an inverse has not been implemented for it.
         """
-        if not self.definition:
+        if self.definition is None:
```

Comment:
What's the difference between `None` and empty in this context?

Reply:
`definition=[]` means an empty circuit:
```
q_0: |0>

q_1: |0>
```

`definition=None` means there is no definition for it and it is unknown how to expand it. i.e. opaque gate.

Reply:
gotcha...